src/elements.cpp

```diff
@@ -388,8 +398,11 @@ G2Element operator*(const G2Element& a, const bn_t& k)
     bn_new(nonConstK[0]);
     bn_copy(nonConstK[0], k);
     g2_mul(ans, nonConstA.q, nonConstK[0]);
+    bn_free(nonConstK[0]);
```

Comment:
my understanding of the relic types, `g1_t`, `ep_t` etc. is that the are pointers to `calloc()`ed memory (allocated by their `*_new()` functions). Allocating the `bn_t` object *itself* in secure memory is most likely pointless, since the *value* is stored on the regular heap.

see [this](https://github.com/relic-toolkit/relic/blob/50f0b836cbf93bc255f41ab730463a38330114b9/include/relic_bn.h#L142).